### PR TITLE
[RM-Networking] approach  "Networking Strategy & Principle"

### PR DESCRIPTION
**Pull request by ASawwaf**
Develop some content for Networking Strategy & Principle

### CODE REVIEW

**Comment by ASawwaf** [comment]
Networking  Scope under CNTT covers :

-	Network Fabric which consists of ( in most cases ) Spine- Leaf Architecture ( 3 Tier or 5 Tier CLOS architecture ) 
-	Automation of the located Fabric ( Programmable Fabric )
-	Variety of Underlay protocols that can be used as a transport layer within the Fabric 
-	Variety of Overlay protocols 
-	SDN integration with Cloud Infrastructure manager 
-	SRIOV workload interworking 
-	DPDK workload interworking 
-	Service Function Channing  (SFC )  and how can be achieved  
-	New Networking technologies ( SmartNIC , FPGA ,..) 
-	Inter K8s POD communication and CNI 

Networking  Strategy & Principles  under CNTT as following :

-	Agnosticity : Build Network Fabric that can carry any type of workload Control, User and Management plan for 
-	Abstraction & Decoupling: The VNF / CNF should be totally decoupled from Network Fabric underneath 
-	Automation: Complete Automation, starting from Fabric provisioning to automated VNF/CNF onboarding

**Comment by ASawwaf** [comment]
@walterkozlowski @peterwoerndle @markshostak @oyayan @rabi-abdel @TFredberg 

Dears, Your comments on above before I add it to the main

**Comment by petorre** [comment]
On

> * Abstraction & Decoupling: The VNF / CNF should be totally decoupled from Network Fabric underneath

This is OK for current NFs and changes if there is any component/algorithm running in Programmable Fabric.
For some previous version I wrote few lines that could be recycled now or later:
- ... may be deployed in many different ways depending on target use case requirements, workload characteristics (different algorithms implementing pipeline steps) and available platforms. This includes network functions running on server nodes with our without programmable HW acceleration, or functions running on a programmable standalone network switch in the network.
- How each layer consumes only exposed resources from the immediate layer below, and try to never jump layers or clearly define CNTT exception policy (including how to handle the introduced breach of layer separation)
- Agree which Functionality and Management API those layers are exposing towards reaching Programmable Fabric
- How to do it in platforms where hardware layers are shared by multiple tenants.

**Comment by walterkozlowski** [comment]
@peterwoerndle @markshostak @oyayan @rabi-abdel @TFredberg @oyayan @ASawwaf @petorre 
I agree with Petar's comment.  We cannot close our eyes on the technology evolution towards programmable fabric happening right now. 

I propose changing "Abstraction & Decoupling: The VNF / CNF should be totally decoupled from Network Fabric underneath" into the one below (borrowing from Petar's comment above):

Abstraction: A standardized layer between Virtualization Layer domains and the Networking Hardware layer resource pool hiding the implementation details of the Networking Hardware. 
Note: This Principle may be deployed in many different ways depending on target use case requirements, workload characteristics (different algorithms implementing pipeline steps) and available platforms. This includes network functions running on server nodes with our without programmable HW acceleration, or functions running on a programmable standalone network switch in the network.

Please comment on the above.

@ASawwaf Some house hygiene issues: This is a good discussion but I thought that we agreed that you with Sukhdev would do clean up and prioritization of the Objectives.  Is this coming in this PR?
Thanks for your comments about the scope but Oya and I proposed the scope in PR #1325 so please add the scope comments there (as others did), so we have them in one place, and can discuss on our Monday meeting.

**Comment by ASawwaf** [comment]
@walterkozlowski @petorre 
thanks for your comment, i will add them 

the above it just initial thought, let lockdown it next call 

@walterkozlowski  i missed but between executive summary and scope, i will move them to the scope PR 

thanks ,

any other thought ? @markshostak @oyayan @sukhdevkapur

**Comment by markshostak** [comment]
@ASawwaf Hi Ahmed, To Walter's point, I think we need to capture the Objectives. Can you update the PR description to indicate specifically what area(s) of content this PR is intended to address? Also, direct that people should be reviewing content in the _comment_, not in the commit.

>
> Network Fabric which consists of ( in most cases ) Spine- Leaf Architecture ( 3 Tier or 5 Tier CLOS architecture )

This seems like an implementation detail. It's true, and it's qualified with most cases, but I'm thinking more focus on the abstraction and functionality provided (the "what"), than on the "how". Perhaps put it in an informative appendix?

>
> Automation of the located Fabric ( Programmable Fabric )

Not sure what located refers to. Automation should be agnostic of the implementation, so no need to reference the type of fabric, as that implies a qualifier.

> SDN integration with Cloud Infrastructure manager

This seems like a key item, along with the abstraction/what is provided at the Workload interface point.

> Variety of Underlay protocols that can be used as a transport layer within the Fabric
> Variety of Overlay protocols
> SRIOV workload interworking
> DPDK workload interworking

Is this intended to be informative or prescriptive content? Hard to review w/o knowing the context.

> Service Function Channing (SFC ) and how can be achieved

Yes, but needs to be described as an abstraction.

> New Networking technologies ( SmartNIC , FPGA ,..)

In the RM, wouldn't this be abstracted into the overlay functionality? In other words, where do we draw the line between abstraction and implementation?

> Inter K8s POD communication and CNI

As an abstraction, or to what level? Can this be provided by the generic networking abstraction, and made specific to K8s in RA-2?

> Networking Strategy & Principles under CNTT as following :
>    Agnosticity : Build Network Fabric that can carry any type of workload Control, User and Management plan for

I'd remove "Build", as we don't build anything in RM. Perhaps replace w/ "Abstract"...

**Comment by walterkozlowski** [comment]
@ASawwaf, i have a problem with this section.  Something seems to be wrong.  Where is the new contents from you, which i think it was discussed?  All I can see is the old list of Objectives with a changed name of the section.  Can you please point us to the relevant contents of "Principles and Strategy"? Thanks.

**Comment by ASawwaf** [comment]
updated @walterkozlowski

File: doc/tech/approach.md [hunk 1]
```diff
@@ -40,7 +40,32 @@ For CNTT, a strategy is needed that affords Operators the performance, flexibili
 > _Placeholder for the Executive Summary (targeting Baldy for first draft). **If you are interested in writing this content, please contact the RM lead.**_
 
 <a name="5.1.2"></a>
-### 5.1.2 Networking Strategy & Principle 
+
+### 5.1.2 Networking Principles & Startegy 
+
+•	Abstraction: A standardized layer between Virtualization Layer domains and the Networking Hardware layer resource pool hiding the implementation details of the Networking Hardware.
```
Comment (CsatariGergely) [inline doc/tech/approach.md]: Please use Markdown way of [bulleted list](https://guides.github.com/features/mastering-markdown/).

Comment (ASawwaf) [reply doc/tech/approach.md]: Done , thx

**Comment by ASawwaf** [comment]
@walterkozlowski based on our 30th - March meeting

**Comment by walterkozlowski** [comment]
@ASawwaf are you still working on this contents?  Let the the team know when is the best time to provide comments/reviews/approvals.  As you remember, we want to have a consolidated text before Monday meeting. Thanks!

**Comment by ASawwaf** [comment]
@walterkozlowski , I am done for this content 
need your support to edit on it @walterkozlowski @markshostak @petorre @sukhdevkapur @oyayan all gents are welcome to edit

**Comment by pgoyal01** [comment]
@ASawwaf some observations:

1. Since the contents are in tech/approach, should the title of this PR be {Tech - XXXX]?
2. Why does the content numbering start with Section "5"?
3. I don't think the 2nd paragraph belongs in the document as it pertains to a "specific point in time"

"Readers be aware, in parallel with the CNTT Networking Strategy, the RI team is implementing networking for use today in labs ...  However, the RI will need to be compatible, meaning, it delivers the network connectivity required by the RA and by RC, even if it does not implement the APIs, topology, encapsulation, etc., that will ultimately be specified herein."

4. The RI will implement the APIs, topology etc. specified in the RAs. RM cannot specify the actual APIs supported by the specific RAs (OpenStack and Kubernetes for example).

**Comment by ASawwaf** [comment]
> 1. Since the contents are in tech/approach, should the title of this PR be {Tech - XXXX]?

Good point, all networking points will be consolidated and after it will be merged under RM as it is RM scope not approach scope

**Comment by ASawwaf** [comment]
> @ASawwaf some observations:
> 
> 1. Since the contents are in tech/approach, should the title of this PR be {Tech - XXXX]?
> 2. Why does the content numbering start with Section "5"?
wired i don't know, let see after @walterkozlowski consolidate 

> 3. I don't think the 2nd paragraph belongs in the document as it pertains to a "specific point in time"
> 
> "Readers be aware, in parallel with the CNTT Networking Strategy, the RI team is implementing networking for use today in labs ... However, the RI will need to be compatible, meaning, it delivers the network connectivity required by the RA and by RC, even if it does not implement the APIs, topology, encapsulation, etc., that will ultimately be specified herein."
> 
> 1. The RI will implement the APIs, topology etc. specified in the RAs. RM cannot specify the actual APIs supported by the specific RAs (OpenStack and Kubernetes for example).

@pgoyal01 valid point and we need to make it clear, any suggestion

**Comment by walterkozlowski** [comment]
I am thinking of removing this approach part altogether. It was written by Mark some time ago and I don’t feel it fits in the current text anyhow!

Walter Kozlowski
Architecture & Strategy Principal
Converged Cloud & Infrastructure Principal
Networks and IT
Telstra
+61 407 724 612
________________________________

**Comment by walterkozlowski** [comment]
All this content is now merged into PR #1325

**Pull request closed without merging**